extensions/objects/templates/ImageWithLabelsEditor.js

```diff
@@ -0,0 +1,358 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Every editor directive should implement an alwaysEditable option. There
+// may be additional customization options for the editor that should be passed
+// in via initArgs.
+oppia.directive('imageWithLabelsEditor', [
```

Comment:
At first glance there seems to be way too much duplication here with the existing image interaction. We'll need to do a bit of work later to reduce that and factor common code out, but let's leave that for a later review when this PR is cleaned up a bit.